libcontainer/apparmor/apparmor_unsupported.go

```diff
@@ -1,5 +1,4 @@
 //go:build !linux
-// +build !linux
```

Comment:
I'm on the fence on already dropping these, as we know libcontainer is still used by external projects, and those may not yet have updated to go 1.18+ (I saw there was some discussion among containerd maintainers as well)

I guess it slightly depends on which release this will get into, and overall it seems ok to (start) dropping 1.17 in CI, but just on the fence on doing a hard _break_ for Go < 1.18

Reply:
The tags removal is done deliberately so the codebase won't work with older golang. This way we can start using new(ish) features of golang 1.18.

I was more conservative about it in the past, supporting three golang releases (including the latest unsupported version), but now when we have branches, we can always say "if you want older golang, use older runc/libcontainer). To me it seems a nice balance between using new golang features and supporting older releases.

So, this code is in main, meaning it's going to be 1.2.0, which we will hopefully release in September/October. From what I see, k8s (I guess this is the biggest user of runc) is switching to newest golang very aggressively.

Reply:
> overall it seems ok to (start) dropping 1.17 in CI, but just on the fence on doing a hard break for Go < 1.18

Doesn't work this way. Once we drop Go 1.17 and someone will add e.g. use of `strings.Cut`, we'll never know that the code is broken for go < 1.18 until someone will use it (after the release). So it's either we support it (and test it in CI), or we don't (and we make this deliberate). Having something in the middle doesn't seem like a good way to me.

Reply:
@thaJeztah PTAL ^^

Reply:
@AkihiroSuda what do you think?

Reply:
My main concern is that `<unknown>` project or distro may still be using older Go versions (I recall discussions about Debian etc not updating).

And while we no longer test on those versions, making things a fully breaking change (not even possible to compile) a bit harsh, and _possibly_ even would require a major version update (from a SemVer perspective); I know Golang's own packages, such as `golang.org/x/sys` continue to include both old and new-style build-tags; https://github.com/golang/sys/blob/1c4a2a72c664c42691e45fe3a31510ec5ba13cfb/unix/str.go#L5-L6

Reply:
Removed the last commit; PTAL @thaJeztah @AkihiroSuda